smacrobs_web/src/main/java/com/web/controller/MainController.java

```diff
@@ -65,22 +72,173 @@ public ModelAndView redirectFromFitbit(
         logger.debug("Successfully redirected from Fitbit "+response);
         FitbitTokens fitbitTokens = this.fitbitOuthService.getFitbitTokens(response);
         logger.debug("FitbitTokens are received!");
+        
         userProfileService.setFitbitTokens(fitbitTokens);
-        UserProfile userProfile = userProfileService.getUserProfileDetails();
+        UserProfile userProfile = userProfileService.getUserDetails();
 
         fitbitDetailsService.setFitbitTokens(fitbitTokens);
+        
         SleepDetails sleepDetails = fitbitDetailsService.getSleepDetails();
-        HeartRateDetails heartRateDetails = fitbitDetailsService
-                .getHeartRateDetails();
+        HeartRateDetails heartRateDetails = fitbitDetailsService.getHeartRateDetails();
+        
+        FoodDetails foodDetails = fitbitDetailsService.getFoodDetails();
+        ActivityDetails activityDetails = fitbitDetailsService.getActivityDetails();
+        WaterDetails waterDetails = fitbitDetailsService.getWaterDetails();
+        ActivityGoalDetails activityGoalDetails = fitbitDetailsService.getActivityGoalDetails();
 
         mv = new ModelAndView("dashboard");
-        mv.addObject("userProfile", userProfile);
+
         mv.addObject("sleepDetails", sleepDetails);
-        mv.addObject("heartRateDetails",heartRateDetails);
+        
+        mv = getHeartRate(mv, heartRateDetails);
+        mv = getActivity(mv,activityDetails);
+        mv = getFood(mv,foodDetails);
+        mv = getWater(mv,waterDetails);
+        mv = getUserProfile(mv, userProfile);
+        mv = getActivityGoals(mv, activityGoalDetails);
+        
+        
         return mv;
     }
 
-    private String getRedirectURL() {
+	private ModelAndView getActivityGoals(ModelAndView modelAndView, ActivityGoalDetails activityGoalDetails) {
+		String distance = "0";
+		String floor = "0";
+		String step = "0";
+		String cal = "0";
+		
+		try {
+			distance = activityGoalDetails.getGoals().getDistance();
+			floor = activityGoalDetails.getGoals().getFloors();
+			step = activityGoalDetails.getGoals().getSteps();
+			cal = activityGoalDetails.getGoals().getCaloriesOut();
+		}catch(Exception ex) {
+			logger.warn(ex.getMessage());
```

Comment:
Print a message along with the exception, something like `logger.warn("Unable to add activity details to model. "+ex)`